src/Parsers/ExpressionListParsers.cpp

```diff
@@ -830,6 +830,16 @@ class FunctionLayer : public Layer
     explicit FunctionLayer(String function_name_, bool allow_function_parameters_ = true)
         : function_name(function_name_), allow_function_parameters(allow_function_parameters_){}
 
+    bool getResult(ASTPtr & node) override
+    {
+        // FunctionLayer can be the only layer in our Layers stack,
+        //  so we need to check that we exited the main cycle properly
+        if (!finished)
+            return false;
```

Comment:
Can this happen for other types of layers?
I see that `finished` is defined in `Layer` class so I'm not sure why not define this check in `Layer::getResult`.

Reply:
Usually, we always check isFinished() for a layer before calling getResult(). The only case where we don't is when some layer is the only one on the stack. Originally this logic was made for ExpressionLayer since its `finished` is never assigned `true` value. But when ParserFunction started using the new expression parser, the same logic started to apply to LayerFunction.

Anyways it is probably a good idea to add it everywhere just in case anyone adds another parser with the same logic.